### ISSUE TITLE
Helm chart CD

### ISSUE THREAD

**Issue by vdice** [spinkube/runtime-class-manager#236]
This project has a pre-existing [Release helm chart](https://github.com/spinkube/runtime-class-manager/actions/workflows/helm-chart-release.yml) workflow, but I don't think it has actually released the chart yet.  Maybe @voigt can double-check?

The workflow uses the [helm/chart-releaser GitHub action](https://github.com/helm/chart-releaser-action) which does the following:

> It does this – during every push to main – by checking each chart in your project, and whenever there's a new chart version, creates a corresponding [GitHub release](https://help.github.com/en/github/administering-a-repository/about-releases) named for the chart version, adds Helm chart artifacts to the release, and creates or updates an index.yaml file with metadata about those releases, which is then hosted on GitHub Pages. You do not need an index.yaml file in main at all because it is managed in the gh-pages branch.

Do we like this process and want to continue using this action?

Another approach, slightly lighter-weight, would be to set up CD to publish the chart as an artifact to the GitHub Container Registry for this repo (eg `ghcr.io/spinkube/charts/runtime-class-manager`) and then users can install via helm using the `oci://` reference functionality, eg `helm install ... oci://ghcr.io/spinkube/charts/runtime-class-manager`.  Then we don't need to host ye olde chart index.yaml or create a GH release to hold artifacts, etc.  (Helm has supported pulling from OCI registries for a few years now, since ~3.8.0)

Thoughts?

**Comment by 0xE282B0** [comment]
Right, the Helm chart releaser was never properly configured in this repo.
See e.g. https://github.com/spinkube/runtime-class-manager/actions/runs/12667742367/job/35301697192
```
Looking up latest tag...
Discovering changed charts since '3f8b0b8dbe42bf31e7a0d54418612c48dcb51b14'...
WARNING: deploy/helm/crds/Chart.yaml is missing, assuming that 'deploy/helm/crds' is not a Helm chart. Skipping.
WARNING: deploy/helm/templates/Chart.yaml is missing, assuming that 'deploy/helm/templates' is not a Helm chart. Skipping.
Nothing to do. No chart changes detected.
```

Pushing it to the GHCR sounds good for me. 👍
